File: scripts/fix_schema.js
const fs = require('fs');

// We don't have glob installed globally maybe, let's just use pure node fs to find files
const path = require('path');

function getFiles(dir, files = []) {
  const fileList = fs.readdirSync(dir);
  for (const file of fileList) {
    const name = `${dir}/${file}`;
    if (fs.statSync(name).isDirectory()) {
      getFiles(name, files);
    } else {
      if (name.endsWith('page.tsx')) {
        files.push(name);
      }
    }
  }
  return files;
}

const files = getFiles('src/app/(public)/fleet');
let count = 0;

for (const f of files) {
  let content = fs.readFileSync(f, 'utf-8');
  let original = content;

  // 1. Add applicableCountry
  content = content.replace(
    /("@type":\s*"MerchantReturnPolicy",\s*)("returnPolicyCategory")/g,
    '$1"applicableCountry": "SA",\n            $2'
  );

  // 2. Change hasCertification to award
  content = content.replace(
    /"hasCertification":\s*"([^"]+)"/g,
    '"award": "$1"'
  );

  if (content !== original) {
    fs.writeFileSync(f, content, 'utf-8');
    count++;
    console.log(`Updated ${f}`);
  }
}

console.log(`Successfully updated ${count} files.`);

File: scripts/fix_schema.py
import re
import glob
import os

files = glob.glob('src/app/(public)/fleet/*/page.tsx')
count = 0

for f in files:
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
    
    original = content
    
    # 1. Add applicableCountry
    content = re.sub(
        r'("@type":\s*"MerchantReturnPolicy",\s*)("returnPolicyCategory")',
        r'\1"applicableCountry": "SA",\n            \2',
        content
    )
    
    # 2. Change hasCertification to award
    content = re.sub(
        r'"hasCertification":\s*"([^"]+)"',
        r'"award": "\1"',
        content
    )
    
    if content != original:
        with open(f, 'w', encoding='utf-8') as file:
            file.write(content)
        count += 1
        print(f"Updated {f}")

print(f"Successfully updated {count} files.")
